components/button/src/dropdown-button/__tests__/dropdown-button.test.js

```diff
@@ -43,6 +44,28 @@ describe('<DropdownButton>', () => {
                     })
                 )
             })
+            it('closes dropdown when escape key is pressed', async () => {
+                const componentText = 'Dropdown Content'
+
+                const { getByTestId, queryByText } = render(
+                    <DropdownButton component={componentText} />
+                )
+
+                const toggleButton = getByTestId(
+                    'dhis2-uicore-dropdownbutton-toggle'
+                )
+                fireEvent.click(toggleButton)
+
+                await waitFor(() => {
+                    expect(queryByText(componentText)).toBeInTheDocument()
+                })
+
+                fireEvent.keyDown(document, { key: 'Escape' })
+
+                await waitFor(() => {
+                    expect(queryByText(componentText)).not.toBeInTheDocument()
+                })
+            })
         })
```

Comment:
Could we have a test that wraps this button in a modal, and make sure that only the dropdown is closed when esc is pressed?